key-rotator/main.go

```diff
@@ -0,0 +1,414 @@
+		c, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+		if err != nil {
+			log.Fatal().Err(err).Msgf("Couldn't get out-of-cluster Kubernetes config: %v", err)
+		}
+		cfg = c
+		log.Info().Msgf("Using out-of-cluster Kubernetes config")
+	}
+
+	k8s, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Couldn't create Kubernetes client: %v", err)
+	}
+	keyStore, err := storage.NewKey(k8s.CoreV1().Secrets(*namespace), *prioEnv)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Couldn't create key store: %v", err)
+	}
+
+	// Get Manifest storage client.
+	log.Info().Msgf("Creating manifest store")
+	var opts []storage.ManifestOption
+	if *awsRegion != "" {
+		opts = append(opts, storage.WithAWSRegion(*awsRegion))
+	}
+	if *awsProfile != "" {
+		opts = append(opts, storage.WithAWSProfile(*awsProfile))
+	}
+	manifestStore, err := storage.NewManifest(ctx, *manifestBucketURL, opts...)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Couldn't create manifest store: %v", err)
+	}
+
+	// ...and go!
+	if *dryRun {
+		log.Info().Msgf("--dry-run is specified: no writes will actually occur")
+		keyStore = dryRunKeyStore{keyStore}
+		manifestStore = dryRunManifestStore{manifestStore}
+	}
+	if err := rotateKeys(ctx, rotateKeysConfig{
+		keyStore:        keyStore,
+		manifestStore:   manifestStore,
+		now:             time.Now(),
+		locality:        *locality,
+		ingestors:       ingestorLst,
+		prioEnvironment: *prioEnv,
+		csrFQDN:         *csrFQDN,
+		batchRotationCFG: key.RotationConfig{
+			CreateKeyFunc:     key.P256.New,
+			CreateMinAge:      *batchSigningKeyCreateMinAge,
+			PrimaryMinAge:     *batchSigningKeyPrimaryMinAge,
+			DeleteMinAge:      *batchSigningKeyDeleteMinAge,
+			DeleteMinKeyCount: *batchSigningKeyDeleteMinCount,
+		},
+		packetRotationCFG: key.RotationConfig{
+			CreateKeyFunc:     key.P256.New,
+			CreateMinAge:      *packetEncryptionKeyCreateMinAge,
+			PrimaryMinAge:     *packetEncryptionKeyPrimaryMinAge,
+			DeleteMinAge:      *packetEncryptionKeyDeleteMinAge,
+			DeleteMinKeyCount: *packetEncryptionKeyDeleteMinCount,
+		},
+	}); err != nil {
+		log.Fatal().Err(err).Msgf("Couldn't rotate keys: %v", err)
+	}
+	log.Info().Msgf("Keys rotated successfully")
+}
+
+type rotateKeysConfig struct {
+	// Dependencies.
+	keyStore      storage.Key
+	manifestStore storage.Manifest
+
+	// Configuration.
+	now               time.Time
+	locality          string
+	ingestors         []string
+	prioEnvironment   string
+	csrFQDN           string
+	batchRotationCFG  key.RotationConfig
+	packetRotationCFG key.RotationConfig
+}
+
+func rotateKeys(ctx context.Context, cfg rotateKeysConfig) error {
+	// Retrieve keys & manifests.
+	log.Info().Msgf("Reading keys & manifests")
+	oldPacketEncryptionKey, oldBatchSigningKeyByIngestor, oldManifestByIngestor, err := readKeysAndManifests(ctx, cfg.keyStore, cfg.manifestStore, cfg.locality, cfg.ingestors)
```

Comment:
nit: wrap long lines

Reply:
Done. I went for a "soft" limit of 100 columns for code (allowing things like human-readable strings etc to scroll off to the right pretty far in some cases), going over slightly if I deemed things more readable that way.